src/examples/3d_fluid_sim/fluid_sim.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#ifndef DIM
+#define DIM 3
+#endif
+
+#if DIM == 2
+#define DIM 3
```

Comment:
I'd prefer to error here if the wrong DIM is specified for this simulation

Reply:
Done. Uses an error directive now.